challengeMk2/ViewModels/StarSystemsViewModel.cs

```diff
@@ -50,28 +52,28 @@ public StarSystem SelectedSystem
 
         async Task LoadSystemData()
         {
-            InitializeDatabase();
+            await InitializeDatabase();
 
             CurrentConnectivity = Connectivity.NetworkAccess;
 
             var expirationDate = Preferences.Get("dbExpirationDate", null);
 
             if (expirationDate != null && DateTime.Now <= DateTime.Parse(expirationDate))
             {
-                DisplaySavedDatas();
+                await DisplaySavedDatas();
             }
             else
             {
                 await RetreiveAndDisplayApiDatas();
             }
         }
 
-        void InitializeDatabase()
+        async Task InitializeDatabase()
         {
             if (App.Database == null)
             {
                 App.Database = new SQLiteDataService();

```

Comment:
You're setting a property of the Application instance ... this is a sign that it probably shouldn't be done here.